### PR TITLE
fix: add uns metadata migration to AnnData to Seurat conversion

### PR DESCRIPTION
**Pull request by metakuni**
### Reviewers
**Functional:** 
@pablo-gar 
@ebezzi 

**Readability:** 
@danieljhegeman 

---

## Changes
- modify: uses an updated `sceasy` which adds migration of the `uns` metadata to the Seurat `misc` slot (see annotations for details)

### CODE REVIEW

File: Dockerfile.processing_base [hunk 1]
```diff
@@ -23,5 +23,5 @@ RUN R -e "BiocManager::install('SingleCellExperiment')"
 #       See https://github.com/cellgeni/sceasy#installation
 RUN R -e "BiocManager::install('LoomExperiment')"
 RUN R -e "install.packages(c('devtools', 'igraph'));"
-RUN R -e "library(devtools); devtools::install_github('cellgeni/sceasy')"
+RUN R -e "library(devtools); devtools::install_github('metakuni/sceasy')"
```
Comment (metakuni) [inline Dockerfile.processing_base:26]: This is the forked repo containing the fix to add the `uns` -> `misc` metadata migration in the AnnData to Seurat conversion function:
* https://github.com/metakuni/sceasy

File: backend/corpora/dataset_processing/make_seurat.R [hunk 2]
```diff
@@ -5,4 +5,9 @@ install_version("SeuratObject", version = "4.0.2", repos = "http://cran.rstudio.
 
 h5adPath <- commandArgs(trailingOnly = TRUE)[1]
 
-sceasy::convertFormat(h5adPath, from="anndata", to="seurat", outFile = gsub(".h5ad", ".rds", h5adPath), main_layer = "data")
+sceasy::convertFormat(h5adPath,
+                      from="anndata",
+                      to="seurat",
+                      outFile = gsub(".h5ad", ".rds", h5adPath),
+                      main_layer = "data",
+                      target_uns_keys = c("schema_version", "title", "X_normalization", "batch_condition", "default_embedding", "X_approximate_distribution"))
```
Comment (metakuni) [inline backend/corpora/dataset_processing/make_seurat.R:13]: The forked repo adds a new optional parameter that specifies the list of keys from the AnnData object's `uns` metadata to migrate to the Seurat object's `misc` slot.

The allowed keys are specified in the curation schemas:
* https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/2.0.0/schema.md#uns-dataset-metadata
* https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/2.0.0/seurat_encoding.md#dataset-metadata

Note, the latter will be updated to reflect the above when this PR is merged:
* https://github.com/chanzuckerberg/single-cell-curation/pull/179

**Comment by metakuni** [comment]
No rush on the review! 😄  Have a nice Thanksgiving weekend, all! 🖖

**Review by ebezzi — APPROVED** [review]

**Comment by metakuni** [comment]
Tested in rdev, uns metadata is 👍

**Pull request merged**